### PR TITLE
Converted fully to QBox + Fixed Flatbed not spawning

### PR DESCRIPTION
**Pull request by ShomyTKD**
## Description

Converted everything to QBox and fixed Flatbed not spawning.

## Checklist

- [X] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [X] My code fits the style guidelines.
- [X] My PR fits the contribution guidelines.

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -1,4 +1,3 @@
-local QBCore = exports['qbx-core']:GetCoreObject()
 local PlayerJob = {}
```
Comment (mafewtm) [inline client/main.lua:1]: This can be removed with all the other PlayerJob instances being changed to QBX.PlayerData.job

File: client/main.lua [hunk 2]
```diff
@@ -215,16 +214,24 @@ end
...
+    TriggerEvent("vehiclekeys:client:SetOwner", plate)
     SetVehicleEngineOn(veh, true, true, false)
     for i = 1, 9, 1 do
         SetVehicleExtra(veh, i, false)
     end
 end)
 
 RegisterNetEvent('QBCore:Client:OnPlayerLoaded', function()
-    PlayerJob = QBCore.Functions.GetPlayerData().job
+    PlayerJob = QBX.PlayerData.job
```
Comment (mafewtm) [inline client/main.lua]: This can be removed

File: client/main.lua [hunk 3]
```diff
@@ -290,23 +297,26 @@ RegisterNetEvent('qb-tow:client:TowVehicle', function()
...
-                            disableMovement = true,
-                            disableCarMovement = true,
-                            disableMouse = false,
-                            disableCombat = true,
-                        }, {
-                            animDict = "mini@repair",
-                            anim = "fixing_a_ped",
-                            flags = 16,
-                        }, {}, {}, function() -- Done
+                        if lib.progressBar({
```
Comment (mafewtm) [inline client/main.lua:298]: Was there a reason for removing the disable movement, mouse, and combat?

File: client/main.lua [hunk 4]
```diff
@@ -318,25 +328,28 @@ RegisterNetEvent('qb-tow:client:TowVehicle', function()
...
-                disableMovement = true,
-                disableCarMovement = true,
-                disableMouse = false,
-                disableCombat = true,
-            }, {
-                animDict = "mini@repair",
-                anim = "fixing_a_ped",
-                flags = 16,
-            }, {}, {}, function() -- Done
+            if lib.progressBar({
```
Comment (mafewtm) [inline client/main.lua:338]: Was there a reason for removing the disable movement, mouse, and combat?

**Comment by mafewtm** [comment]
Fix your linting issues as well. Don't worry about the first 4 regarding BoxZones or combo zones, but the other ones need to be tended to

File: client/main.lua (hunk 3 above)
Comment (ShomyTKD) [reply client/main.lua:298]: Yeah, the fact that you have to be in the vehicle in order to attach the car.

image

File: client/main.lua (hunk 4 above)
Comment (ShomyTKD) [reply client/main.lua:338]: Same here, you must be in a vehicle.

image

**Comment by ShomyTKD** [comment]
> Fix your linting issues as well. Don't worry about the first 4 regarding BoxZones or combo zones, but the other ones need to be tended to

Would here only be needed to remove in first line = 1000?

```
CreateThread(function()
    local sleep = 1000
    while true do
        sleep = 1000
        if showMarker then
            sleep = 0
            DrawMarker(2, Config.Locations["vehicle"].coords.x, Config.Locations["vehicle"].coords.y, Config.Locations["vehicle"].coords.z, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.2, 0.15, 200, 0, 0, 222, false, false, 0, true, false, false, false)
        end
        Wait(sleep)
    end
end)
```

File: client/main.lua (hunk 3 above)
Comment (mafewtm) [reply client/main.lua:298]: That makes no sense as to why they were removed

**Comment by mafewtm** [comment]
> > Fix your linting issues as well. Don't worry about the first 4 regarding BoxZones or combo zones, but the other ones need to be tended to
> 
> Would here only be needed to remove in first line = 1000?
> 
> ```
> CreateThread(function()
>     local sleep = 1000
>     while true do
>         sleep = 1000
>         if showMarker then
>             sleep = 0
>             DrawMarker(2, Config.Locations["vehicle"].coords.x, Config.Locations["vehicle"].coords.y, Config.Locations["vehicle"].coords.z, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.2, 0.15, 200, 0, 0, 222, false, false, 0, true, false, false, false)
>         end
>         Wait(sleep)
>     end
> end)
> ```

Adjust it to this:

```
CreateThread(function()
    local sleep
    while true do
        sleep = 1000
        if showMarker then
            sleep = 0
            DrawMarker(2, Config.Locations["vehicle"].coords.x, Config.Locations["vehicle"].coords.y, Config.Locations["vehicle"].coords.z, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.2, 0.15, 200, 0, 0, 222, false, false, 0, true, false, false, false)
        end
        Wait(sleep)
    end
end)
```

File: client/main.lua (hunk 3 above)
Comment (ShomyTKD) [reply client/main.lua:298]: Yeah well, that's how it was by default, I didn't change that. You can't leave the vehicle once the progressbar starts, so disabling anything is pointless.

**Comment by ShomyTKD** [comment]
All linting fixed (except BoxZones).

**Pull request merged**